rfcs/proposed/iterators_pass_directly_customization/README.md

```diff
@@ -0,0 +1,174 @@
+namespace creates maintenance problems. It either requires lots of closing of nested namespaces, opening of the external
+ library's namespace for the specialization, or it requires separating these specializations to a separate location
+ removed from the types they are specializing for. oneDPL has chosen to use the latter, which can be seen in
+ `include/oneapi/dpl/pstl/hetero/dpcpp/sycl_traits.h`. This has led to several errors where changes to structures should
+  have included changes to sycl_traits but did not, and needed to be fixed later.
+
+In an effort to avoid this same issue for our users, we propose a similar method but instead with a constexpr
+customization point, allowing the user to override that customization point within their own namespace as a free
+function.
+
+### Require Specifically Named Typedef / Using in User's Type
+oneDPL could make official our requirements for users' types to include a typedef or using statement to define if the
+type is "passed directly" like `using is_passed_directly = std::true_type;`, where the absence of this would be
+equivalent to a `std::false_type`.
+
+However, this clutters the user type definitions with specifics of oneDPL. It also may not be as clear what this
+signifies for maintenance of user code without appropriate comments describing the details of oneDPL and SYCL. Users
+have expressed that this is undesirable.
+
+### Wrapper Class
+oneDPL could provide some wrapper iterator `direct_iterator` which wraps an arbitrary base iterator and marks it as
+"passed directly". `direct_iterator` could utilize either of the above alternatives to accomplish this and signal that
+the iterator should be "passed directly". It would need to pass through all operations to the wrapped base iterator and
+make sure no overhead is added in its usage.
+
+There is some complexity in adding such a wrapper iterator, and it would need to be considered carefully to make sure no
+ problems would be introduced. This wrapper class may obfuscate users' types and make them more unwieldy to use. It is
+ also less expressive than the other options in that it only has the ability to unilaterally mark a type as "passed
+ directly". There is no logic that can be used to express some iterator type which may be conditionally "passed
+ directly", other than to have logic to conditionally apply the wrapper in the first place. This option seems less clear
+ and has more opportunity to cause problems.
+
+## Testing
+We will need a detailed test checking both positive and negative responses to
+`oneapi::dpl::is_passed_directly_in_onedpl_device_policies_v` to ensure they have the expected result. This should
+include tests with custom types, combinations of iterators, USM pointers, etc.
+
+## Open Questions
+
+* Is there a better, more concise name than `oneapi::dpl::is_passed_directly_in_onedpl_device_policies[_v]` that
+properly conveys the meaning to the users?
+    * Other names proposed:
```

Comment:
`oneapi::dpl::is_passed_directly_to_device` ?

Reply:
I added the two proposed names you raised and a note to the open questions section.

Reply:
I've separated the names of the ADL customization point and the public trait.  This allows us to provide both the struct and value version of the trait, which is the norm.